packages/edit-site/src/components/page-patterns/index.js

```diff
@@ -351,9 +351,12 @@ export default function DataviewsPatterns() {
 	// Reset the page number when the category changes.
 	useEffect( () => {
 		if ( previousCategoryId !== categoryId ) {
-			setView( DEFAULT_VIEW );
+			setView( {
+				...view,
+				page: 1,
+			} );
 		}
-	}, [ categoryId, previousCategoryId ] );
+	}, [ categoryId, previousCategoryId, view ] );
```

Comment:
Passing the same state as a dependency as the effect updates could easily lead to infinite loops, especially when this state is an object. Only having a category check prevents that here.

Let's use the [state updater function](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state) instead - `setView( ( prevView ) => ( { ...prevView, page: 1 } ) );`.

Reply:
Updated in [0383686](https://github.com/WordPress/gutenberg/pull/63711/commits/03836866a900df683edb5cb5c04e0b27e315dd44).